### PR TITLE
Basic Solutions Component

### PR DESCRIPTION
**Pull request by dagrooms52**
- Pre-built restart solution allows a full app restart over ARM API
- Solutions can only be added to Insights currently
- Solutions aren't very customizable currently, but I would like to add the pre-built Restart solution to a few detectors to get early feedback from customers

### CODE REVIEW

File: AngularApp/projects/diagnostic-data/src/lib/services/diagnostic-site.service.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    parseResourceUri(resourceUri: string): any {
+        return null;
+    }
+
+    restartSite(subscriptionId: string, resourceGroup: string, siteName: string): Observable<boolean> {
+        return null;
+    }
+
+    restartSiteFromUri(resourceUri: string): Observable<boolean> {
+        console.log("Fake site restart called")
```
Comment (cindylovescoding) [inline AngularApp/projects/diagnostic-data/src/lib/services/diagnostic-site.service.ts]: Can we remove all these console logs?

File: AngularApp/projects/diagnostic-data/src/lib/components/solutions/solutions.component.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+@Component({
+  selector: 'solutions',
+  templateUrl: './solutions.component.html',
+  styleUrls: ['./solutions.component.scss']
+})
+export class SolutionsComponent implements OnInit {
+
+  @Input() solutions: Solution[];
+  selectedSolution = 0;
```
Comment (stevenernst131) [inline AngularApp/projects/diagnostic-data/src/lib/components/solutions/solutions.component.ts]: I know I changed this :) but just choose between using selectedSolution index of selected object and remove the other

**Comment by stevenernst131** [comment]
Do you have some screenshots?

**Comment by dagrooms52** [comment]
A couple things I need to do before merging:
 - add logging for actions
 - get a real response from ARM API (currently returns false even on success)

I'll also add some screenshots to the PR

**Comment by puneetg1983** [comment]
hi @dagrooms52 , please update your PR with some screenshots...

**Comment by dagrooms52** [comment]
Internal view of a Solution with the option to copy instructions to send to the customer including a direct link to the Detector:
solution_internal_view

The external (customer) view of the same Solution as above:
solution_external_view

A Solution without warnings for the action, to be used on Solutions that aren't potentially harmful to the customer's application. This also shows the "running" feedback that the customer will see when they invoke the action:
solution_external_no_confirm

**Comment by stevenernst131** [comment]
Please select Squash and Merge when you merge your changes :)

**Comment by dagrooms52** [comment]
@stevenernst131 We should be able to set Squash and Merge as the default according to this doc: https://help.github.com/en/articles/configuring-commit-squashing-for-pull-requests

**Pull request merged**